lib/auth/sessions.go

```diff
@@ -623,6 +624,12 @@ func (a *Server) CreateAppSessionFromReq(ctx context.Context, req NewAppSessionR
 			AppPublicAddr: req.PublicAddr,
 			AppName:       req.AppName,
 		},
+		AWSSessionMetadata: apievents.AWSSessionMetadata{
+			// Make an educated guess of the role session name. We have to
+			// guess here since this event is emitted on Auth instead of the
+			// App service that issues the actual AssumeRole call.
+			AWSRoleSessionName: guessAWSRoleSessionName(clusterName.GetClusterName(), req),
```

Comment:
Guessing a value for audit makes me uneasy. The docs on AWSSessionMetadata make no mention of this being guessed either.